Mathlib/Algebra/MvPolynomial/Equiv.lean

```diff
@@ -318,40 +318,257 @@ lemma optionEquivRight_X_none : optionEquivRight R S₁ (X none) = C Polynomial.
 lemma optionEquivRight_C (r : R) : optionEquivRight R S₁ (C r) = C (Polynomial.C r) := by
   simp only [optionEquivRight_apply, aeval_C, algebraMap_apply, Polynomial.algebraMap_eq]
 
+section FinSuccEquivNth
+
+variable {n : ℕ} (p : Fin (n + 1))
+
+/-- The algebra isomorphism between multivariable polynomials in `Fin (n + 1)` and polynomials over
+  multivariable polynomials in `Fin n`, where the `p`-th (pivot) variable is the indeterminate `X`.
+
+  `finSuccEquiv` is the special case when `p = 0`. -/
+def finSuccEquivNth : MvPolynomial (Fin (n + 1)) R ≃ₐ[R] Polynomial (MvPolynomial (Fin n) R) :=
+  (renameEquiv R (_root_.finSuccEquiv' p)).trans (optionEquivLeft R (Fin n))
+
+theorem finSuccEquivNth_eq :
+    (finSuccEquivNth R p : MvPolynomial (Fin (n + 1)) R →+* Polynomial (MvPolynomial (Fin n) R)) =
+      eval₂Hom (Polynomial.C.comp (C : R →+* MvPolynomial (Fin n) R))
+        (Fin.insertNth p Polynomial.X (Polynomial.C ∘ X)) := by
+  ext j : 2
+  · simp only [finSuccEquivNth, optionEquivLeft_apply, aeval_C, AlgEquiv.coe_trans, RingHom.coe_coe,
+      coe_eval₂Hom, comp_apply, renameEquiv_apply, eval₂_C, RingHom.coe_comp, rename_C]
+    rfl
+  · refine Fin.succAboveCases p ?_ ?_ j <;> simp [finSuccEquivNth]
+
+theorem finSuccEquivNth_apply (f : MvPolynomial (Fin (n + 1)) R) :
+    finSuccEquivNth R p f =
+      eval₂Hom (Polynomial.C.comp (C : R →+* MvPolynomial (Fin n) R))
+        (Fin.insertNth p Polynomial.X (Polynomial.C ∘ X)) f := by
+  rw [← finSuccEquivNth_eq, RingHom.coe_coe]
+
+theorem finSuccEquivNth_comp_C_eq_C :
+    (↑(finSuccEquivNth R p).symm : Polynomial (MvPolynomial (Fin n) R) →+* _).comp
+        (Polynomial.C.comp C) = (C : R →+* MvPolynomial (Fin n.succ) R) := by
+  refine RingHom.ext fun x => ?_
+  rw [RingHom.comp_apply]
+  refine
+    (finSuccEquivNth R p).injective
+      (Trans.trans ((finSuccEquivNth R p).apply_symm_apply _) ?_)
+  simp only [finSuccEquivNth_apply, MvPolynomial.eval₂Hom_C]
+
+variable {R} {p}
+
+theorem finSuccEquivNth_X_same : finSuccEquivNth R p (X p) = Polynomial.X := by
```

Comment:
Maybe this could be a `simp` lemma?

```suggestion
@[simp]
theorem finSuccEquivNth_X_same : finSuccEquivNth R p (X p) = Polynomial.X := by
```

Also the existing `finSuccEquiv_X_zero` and `finSuccEquiv_X_succ` (not sure),